examples/coalton-json/tests/tests.lisp

```diff
@@ -33,7 +33,16 @@
            (coalton-json:parse-json "[[]]"))
   (is-type 'coalton-json::json/json-array
            (coalton-json:parse-json "[1, 1.0, true, false, [], \"x\"]"))
+  (is-type 'coalton-json::json/json-array
+           (coalton-json::parse-json 
+            (coalton-json::json->string 
+             (coalton-json:parse-json "[1, 1.0, true, false, [], \"x\"]"))))
   (is-type 'coalton-json::json/json-object
            (coalton-json:parse-json "{}"))
   (is-type 'coalton-json::json/json-object
-           (coalton-json:parse-json "{\"x\":0, \"y\":{\"z\":\"z\"}}")))
+           (coalton-json:parse-json "{\"x\":0, \"y\":{\"z\":\"z\"}}"))
+  (is-type 'coalton-json::json/json-object
+           (coalton-json::parse-json 
+            (coalton-json::json->string
+             (coalton-json:parse-json "{\"x\":0, \"y\":{\"z\":\"z\"}}")))))
```

Comment:
I'd like to see some tests here that compare the output off `json->string` to an expected string, for representatives of at least integers (small positive, large positive, small negative, large negative, zero, perhaps), floats (positive and negative zero, very large positive and negative numbers, very small positive and negative numbers, and numbers with non-zero digits to both the left and right of the decimal point, perhaps), booleans (both of em), and strings (idk, whatever). It would also be nice to demonstrate, for types with `Eq` instances, that `(compose parse-json json->string)` is like `id`.

Reply:
added more tests but unfortunately can't do `(compose parse-json json->string)` is like `id` because of the formatting of floats like 1 to 1.0.